File: services/consultationservice/internal/newsfeed/group/domain/group.go
package domain

import (
	"time"

	"go.mongodb.org/mongo-driver/bson/primitive"
)

type Group struct {
	ID          primitive.ObjectID `bson:"_id,omitempty" json:"id"`
	Name        string             `bson:"name" json:"name"`
	Description string             `bson:"description" json:"description"`
	Category    string             `bson:"category" json:"category"` // e.g., "Depression", "Anxiety", "General"
	Icon        string             `bson:"icon" json:"icon"`         // URL or icon identifier
	CreatorID   string             `bson:"creator_id" json:"creator_id"`
	MemberCount int                `bson:"member_count" json:"member_count"`
	IsPublic    bool               `bson:"is_public" json:"is_public"`

	CreatedAt time.Time `bson:"created_at" json:"created_at"`
	UpdatedAt time.Time `bson:"updated_at" json:"updated_at"`
}

type GroupMember struct {
	ID       primitive.ObjectID `bson:"_id,omitempty" json:"id"`
	GroupID  primitive.ObjectID `bson:"group_id" json:"group_id"`
	UserID   string             `bson:"user_id" json:"user_id"`
	Role     string             `bson:"role" json:"role"` // "admin", "member"
	JoinedAt time.Time          `bson:"joined_at" json:"joined_at"`
}

File: services/mobileflutter/mobileapp/lib/models/group.dart
class Group {
  final String id;
  final String name;
  final String description;
  final String category;
  final int memberCount;
  final bool isPublic;
  final String createdAt;

  Group({
    required this.id,
    required this.name,
    required this.description,
    required this.category,
    required this.memberCount,
    required this.isPublic,
    required this.createdAt,
  });

  factory Group.fromJson(Map<String, dynamic> json) {
    return Group(
      id: json['id'] ?? json['_id'] ?? '',
      name: json['name'] ?? '',
      description: json['description'] ?? '',
      category: json['category'] ?? '',
      memberCount: json['memberCount'] ?? json['member_count'] ?? 0,
      isPublic: json['isPublic'] ?? json['is_public'] ?? true,
      createdAt: json['createdAt'] ?? json['created_at'] ?? '',
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'name': name,
      'description': description,
      'category': category,
      'memberCount': memberCount,
      'isPublic': isPublic,
      'createdAt': createdAt,
    };
  }
}
